ngraph/core/src/function.cpp

```diff
@@ -347,6 +384,37 @@ bool Function::visit_attributes(AttributeVisitor& visitor)
     return true;
 }
 
+void Function::add_sinks(const SinkVector& sinks)
+{
+    m_sinks.insert(m_sinks.end(), sinks.begin(), sinks.end());
+    validate_nodes_and_infer_types();
+}
+
+void Function::remove_sink(const std::shared_ptr<op::Sink>& sink)
+{
+    m_sinks.erase(std::remove_if(m_sinks.begin(),
+                                 m_sinks.end(),
+                                 [&sink](std::shared_ptr<op::Sink>& s) { return s == sink; }),
+                  m_sinks.end());
+    validate_nodes_and_infer_types();
```

Comment:
Why do we need to re-validate a function when removing sinks? No new nodes are expected for validation.

Reply:
I think about cases when removing output from operation can be a problem. For example, if we remove sink/result from 1 of Split branches, then IE will not infer it. But I am not sure if validate... can find out such cases.

Reply:
@sadolini We do not guarantee the correct work of the network after deleting the result (or any other node). If we get an inconsistency network, we will crash anyway, whether we call `validate_and_infer `function or not. But the call of `validate_and_infer` function takes a long time, so please remove it from add/remove result/sink functions.

Reply:
Ok, I will remove validate_... from remove methods. But in Add methods we add new nodes in graph (the whole subgraph connected with Assign\Result)  that was not validated on Function creation because we couldn't access it. By the way, could we validate only subgraph that was not validated before somehow if it takes so long time?

Reply:
New nodes will be validated upon creation (ctor of node usually call `validate_and_infer_type `method for this node, except Loop and TensorIterator). So we connect a fully validated subgraph.